src/themes/greyVest/ErrorList.js

```diff
@@ -1,20 +1,11 @@
 import React from 'react'
 import _ from 'lodash/fp'
-import ErrorBlock from './ErrorBlock'
 import ErrorText from './ErrorText'
 
-let ErrorList = ({ ErrorComponent, block = false, children, ...props }) => {
-  if (!ErrorComponent) {
-    ErrorComponent = block ? ErrorBlock : ErrorText
-  }
-  return _.map(
-    e =>
-      e ? (
-        <ErrorComponent key={e} {...props}>
-          {e}
-        </ErrorComponent>
-      ) : null,
-    _.castArray(children)
-  )
-}
+let ErrorList = ({ ErrorComponent = ErrorText, children, ...props }) => _.map(
```

Comment:
Sorry for the churn:
- Use `React.Children.map` and you won't need `_.castArray`.
- No need to `e ? <ErrorComponent ...> : null`. `e && <ErrorComponent />` works
- I wonder if `Component` is better than `ErrorComponent`. After all, `ErrorList` already hints that we're dealing with errors. What do you think?

Reply:
- I changed it back to `_.castArray` when I found out that `React.Children.map` flattens child arrays, such that this will render three ErrorBlocks, rather than two (with the first containing two errors):
  ```
  <ErrorList Component={ErrorBlock}>
    {[['Error 1A', 'Error 1B'], 'Error 2']}
  </ErrorList>
  ```
  Though, that's kind of a silly edge case anyway, and `React.Children.map` is probably safer than `_.castArray`, so I could go either way - might as well use it.

- I didn't think so either, but apparently it is necessary - React complains if you don't explicitly return `null` from a component that doesn't render anything (which is the case for an ErrorList that receives no errors).

- Good point, changed 🙂